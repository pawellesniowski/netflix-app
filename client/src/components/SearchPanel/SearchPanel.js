import React from 'react';
import Button from "../ui-elements/Button/Button";
import './SearchPanel.scss';

const SearchPanel = () => {
  return (
    <div className="search-panel" data-cy="search-panel">
      <label htmlFor="titleInput" data-cy="search-input">
        <span className="label-text">Find Your Movie</span>
        <input className="input" placeholder="Find Your Movie"/>
      </label>
      <div className="search-option-container" data-cy="search-option-container">
        <span className="search-option-text">Search by</span>
        <Button message="title" mode="default"/>
        <Button message="genre" mode="default"/>
        <Button message="search" mode="submit"/>
      </div>
    </div>
  );
};

export default SearchPanel;
