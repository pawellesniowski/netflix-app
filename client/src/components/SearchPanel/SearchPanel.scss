@import "../../common/styles/colors.scss";
@import "../../common/styles/mixins.scss";


.search-panel {
  @include contentContainer;
  background: linear-gradient($color-grey-black, rgba(0, 0, 0, 0.6)), url('../../common/images/netflix-background.jpg');
  color: $color-grey-mercury;
  padding-top: 5rem;
  padding-bottom: 5rem;

  .label-text {
    padding-top: 3rem;
    text-transform: uppercase;
    display: block;
    font-size: 3rem;
    margin-bottom: 0.2rem;
  }

  .input {
    width: 100%;
    padding: 12px 20px;
    color: $color-grey-mercury;
    background-color: transparent;
    border: none;
    border-bottom: 2px solid $color-netflix-red;
    margin-bottom: 1rem;

    &:focus {
      outline: none !important;
      box-shadow: 0 0 10px $color-netflix-red;
    }
  }

  .search-option-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    padding-bottom: 2rem;

    .search-option-text {
      margin: 4px 10px 0 0;
      font-size: 2rem;
    }

    .search-radio-buttons {
      margin: 4px;

      label {
        float: left;
        width: 120px;
        height: 40px;
        margin: 0 4px;
        border-radius: 4px;
        border: 1px solid $color-netflix-red;
        overflow: auto;

        &:hover {
          box-shadow: 0 0 10px $color-netflix-red;
        }

        span {
          text-align: center;
          font-size: 2rem;
          display: block;
          color: $color-grey-mid-grey;
          line-height: 38px;
        }

        input {
          position: absolute;
          top: -20px;
        }

        input:checked + span {
          color: $color-netflix-red;
        }

      }


    }

    .button:not(:last-child) {
      margin-left: 1rem;
    }

    .button:last-child {
      margin-left: auto;
    }
  }
}

