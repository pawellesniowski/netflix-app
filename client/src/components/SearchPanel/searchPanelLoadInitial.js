const LOAD_INITIAL = 'SearchPanel: loading initial values for form';

const loadInitialReducer = (state = {}, action) => {
    switch (action.type) {
        case LOAD_INITIAL:
            return {
                data: action.data,
            };
        default:
            return state;
    }
};

export const loadInitial = data => ({ type: LOAD_INITIAL, data });

export default loadInitialReducer;
