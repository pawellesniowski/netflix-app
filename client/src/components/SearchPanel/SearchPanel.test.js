import React from 'react';
import {shallow} from "enzyme/build";
import {SearchPanel} from "./SearchPanel";

describe("SearchPanel Component", () => {
    let component;
    beforeEach(() => {
        component = shallow(<SearchPanel/>);
    });
    it("Matches the snapshot", () => {
        expect(component).toMatchSnapshot();
    });
});
