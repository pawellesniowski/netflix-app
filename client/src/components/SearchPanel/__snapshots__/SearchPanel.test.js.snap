// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel Component Matches the snapshot 1`] = `ShallowWrapper {}`;
